tests/std/tests/P0356R5_bind_front/test.cpp

```diff
@@ -91,6 +92,42 @@ int main() {
     bound1(4);
     assert(value == 234);
 
+    // Test "perfect forwarding call wrapper" behavior.
+    auto bound5 = bind_front(DetectQualifiers{});
+    assert(bound5() == "modifiable lvalue");
+    assert(as_const(bound5)() == "const lvalue");
+    assert(move(bound5)() == "modifiable rvalue");
+    assert(move(as_const(bound5))() == "const rvalue");
+
+    // Test decay when binding.
+    const int arr[] = {11, 22, 33};
+    const int three = 3;
+
+    auto bound8 = bind_front(
+        [](auto&& a, auto&& f, auto&& i) {
+            using FP = int (*)(int);
+            return is_same_v<decltype(a), const int*&> && is_same_v<decltype(f), FP&> && is_same_v<decltype(i), int&>;
+        },
+        arr, f1, three);
+    assert(bound8());
+
+    // Test forward when calling.
+    auto bound9 = bind_front([](auto&& a1, auto&& a2, auto&& a3, auto&& a4) {
+        constexpr bool same1 = is_same_v<decltype(a1), int&>;
+        constexpr bool same2 = is_same_v<decltype(a2), const int&>;
+        constexpr bool same3 = is_same_v<decltype(a3), int&&>;
+        constexpr bool same4 = is_same_v<decltype(a4), const int&&>;
+        return same1 && same2 && same3 && same4;
+    });
+    assert(bound9(value, three, 1729, move(three)));
+
+    return true;
+}
+
+int main() {
+    assert(test_constexpr());
+    static_assert(test_constexpr());
+
     // Test PMFs.

```

Comment:
Why aren't we testing this PMF section in `test_constexpr`?

Reply:
Because `Cat` is not `constexpr` as it uses `std::string`. It can be moved later when `std::string` is `constexpr`.

Reply:
@CaseyCarter I have this comment on my radar as I'm working on `constexpr string`.  Would it be a good idea to file an issue to track this and further `constexpr`ize this test once [P0980](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p0980r1.pdf) is implemented?